File: LeetCode/p55.cpp
class Solution {
public:
    bool canJump(const vector<int> &nums) {
        int black = 0;
        for (int i = nums.size() - 1; i >= 0; --i) {
            if (nums[i] - black <= 0) {
                if (i == 0)
                    return false;
                ++black;
            } else {
                black = 0;
            }
        }
        return true;
    }
};

File: LeetCode/p55.rs
impl Solution {
  pub fn can_jump(nums: Vec<i32>) -> bool {
    let mut bounder = 0;
    for i in 0..nums.len() {
      if i > bounder {
        return false;
      }
      bounder = bounder.max(i + nums[i] as usize);
    }
    true
  }
}

